sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNodeSuite.scala

```diff
@@ -38,21 +40,32 @@ class SqlScriptingExecutionNodeSuite extends SparkFunSuite {
 
   case class TestBody(statements: Seq[CompoundStatementExec]) extends CompoundBodyExec(statements)
 
-  case class TestSparkStatementWithPlan(testVal: String)
+  case class TestIfElse(
+      conditions: Seq[SingleStatementExec],
+      conditionalBodies: Seq[TestBody],
+      elseBody: Option[TestBody],
```

Comment:
why do we need `TestIfElse`? Just to change the declared type of `conditionalBodies` and `elseBody`?

Reply:
good catch - it's simply a leftover from before before we simplified tests a lot... I've actually removed all `Test*` implementations of non leaf statements - we need to override only leaf statements for testing purposes. Abstractions are done properly, so real implementations of execution nodes from `SqlScriptingExecutionNode` can handle dummy implementations of leaf statements in tests. Should be really clean now, I think...